FueledUtils/CoalescingAction.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  CoalescingAction.swift
+//  FueledUtils
+//
+//  Created by Stéphane Copin on 4/22/16.
+//  Copyright © 2016 Fueled. All rights reserved.
+//
+
+import ReactiveSwift
+import Result
+
+///
+/// Similar to `Action`, except if the action is already executing, subsequent `apply()` call will not fail,
+/// and will be completed with the same output when the initial executing action completes.
+/// Disposing any of the `SignalProducer` returned by 'apply()` will cancel the action.
+///
+public class CoalescingAction<Input, Output, Error: Swift.Error>: ActionProtocol {
+	public typealias OutputType = Output
+	public typealias InputType = Input
+	public typealias ErrorType = Error
+	public typealias ApplyErrorType = Error
```

Comment:
Is there a way to mask the generic type with the associated type, so their names stay the same? (I haven't tried anything here)